src/rdfTypes/GeometryInfoHelpersImpl.h

```diff
@@ -38,37 +38,44 @@ inline ParseResult parseWkt(const std::string_view& wkt) {
   auto wktLiteral = removeDatatype(wkt);
   std::optional<ParsedWkt> parsed = std::nullopt;
   auto type = getWKTType(wktLiteral);
-  switch (type) {
-    case WKTType::POINT: {
-      parsed = pointFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::LINESTRING: {
-      parsed = lineFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::POLYGON: {
-      parsed = polygonFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::MULTIPOINT: {
-      parsed = multiPointFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::MULTILINESTRING: {
-      parsed = multiLineFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::MULTIPOLYGON: {
-      parsed = multiPolygonFromWKT<CoordType>(wktLiteral);
-      break;
-    }
-    case WKTType::COLLECTION: {
-      parsed = collectionFromWKT<CoordType>(wktLiteral);
-      break;
+  using enum WKTType;
+  try {
+    switch (type) {
+      case POINT: {
+        parsed = pointFromWKT<CoordType>(wktLiteral);
+        break;
+      }
+      case LINESTRING: {
+        parsed = lineFromWKT<CoordType>(wktLiteral);
+        break;
+      }
```

Comment:
And can you remove those unneeded `{}` around the cases, they break the coverage unfortunately.